cluster-autoscaler/cloudprovider/gce/gce_url.go

```diff
@@ -72,8 +73,14 @@ func GenerateMigUrl(ref GceRef) string {
 	return fmt.Sprintf(migUrlTemplate, ref.Project, ref.Zone, ref.Name)
 }
 
+// parseGceUrl parses a GCE Compute API URL into the project ID, zone name and resource name.
 func parseGceUrl(url, expectedResource string) (project string, zone string, name string, err error) {
-	errMsg := fmt.Errorf("Wrong url: expected format https://content.googleapis.com/compute/v1/projects/<project-id>/zones/<zone>/%s/<name>, got %s", expectedResource, url)
+	errMsg := fmt.Errorf("Wrong URL: expected format https://www.googleapis.com/compute/v1/projects/<project-id>/zones/<zone>/%s/<name>, got %s", expectedResource, url)
+
+	if expectedResource == "" {
```

Comment:
This check is IMO redundant, but if we're going to keep it, let's rephrase the message to sth more like wrong resource error message. Logging what we got and what we expected is usually a lot more useful than a generic error message (which is completely different from anything else we have in the code, btw, this is not user-facing API). 

Also, it seems errors.New() and fmt.Errorf() are basically identical, can we stick to using just one everywhere?

Reply:
Noted RE: error message intent.

- I explicitly did not use `fmt.Errorf` as I wasn't using a format string, and that'd be a linter error.
- I'll add an explicit check for `instances` and `instanceGroups`
- Likely, I'll take this opportunity to improve the URL parsing via `net/url.Parse` so that we can correctly validate scheme vs. host vs. path components.